vcxagency-node/src/scripts/download-certs.js

```diff
@@ -0,0 +1,44 @@
+const { S3Client, GetObjectCommand } = require('@aws-sdk/client-s3')
+const fs = require('fs')
+const logger = require('../logging/logger-builder')(__filename)
+
+module.exports.fetchAwsAssets = async function fetchAwsAssets (appConfig) {
+  if (appConfig.SERVER_ENABLE_TLS === 'false') {
+    logger.info('TLS disabled, skipping AWS assets download...')
+    return
+  }
+
+  const certPath = appConfig.CERTIFICATE_PATH
+  const keyPath = appConfig.CERTIFICATE_KEY_PATH
+  const isCertAvailable = (appConfig.AWS_BUCKET_NAME && appConfig.AWS_DOMAIN_NAME) || (fs.existsSync(certPath) && fs.existsSync(keyPath))
+
+  if (!isCertAvailable) {
+    throw Error(`TLS enabled, but AWS S3 bucket or domain name not defined, or no certificate already found on path ${certPath} or no key found on path ${keyPath}`)
+  }
+
+  logger.info('Downloading S3 assets')
+  await fetchAwsAsset(appConfig.AWS_BUCKET_NAME, appConfig.AWS_DOMAIN_NAME + '.crt', certPath)
```

Comment:
I'd rather go for 
`AWS_S3_CERT_PATH`
`AWS_S3_CERT_KEY_PATH`
It's easier to understand and is more flexible

Reply:
I think `AWS_S3_CERT_BUCKET_NAME` and `AWS_S3_CERT_BUCKET_KEY_BASE` is more accurate.

Reply:
I would prefer to designate just individual bucket path for cert+key. It puts weaker contract on configuration and S3 structure. To verify that path is correct, I take a look at path and see if it exists. To check if these variables are correct, in my head I need to do 3 concatenations first.
Seems like somewhat of a personal preference matter, I've left my opinion, but leaving the final choice up to you.

Reply:
`bucketname/key` it is